pkg/kritis/review/validating_transport.go

```diff
@@ -101,51 +103,74 @@ func (avt *AttestorValidatingTransport) validatePublicKeyId(pubKey v1beta1.Publi
 	return nil
 }
 
+// NOTE: I am considering changing this function signature to:
+// func (avt *AttestorValidatingTransport) CheckValidatedAttestation(image string) error
+// The function would propagate the error (or success) of the
+// verifier.VerifyAttestation call. It currently returns ValdiatedAttestations,
+// which contain unused and not particularly meaningful information.
+// This is not a trivial change, and would be done in a follow-up PR.
 func (avt *AttestorValidatingTransport) GetValidatedAttestations(image string) ([]attestation.ValidatedAttestation, error) {
-	keys := map[string]v1beta1.PublicKey{}
+	publicKeys := []cryptolib.PublicKey{}
 	numKeys := len(avt.Attestor.Spec.PublicKeys)
-	for i, pubKey := range avt.Attestor.Spec.PublicKeys {
-		if err := avt.validatePublicKey(pubKey); err != nil {
+	validatedAtts := []attestation.ValidatedAttestation{}
+
+	for i, attestorKey := range avt.Attestor.Spec.PublicKeys {
+		if err := avt.validatePublicKey(attestorKey); err != nil {
 			// warning level because single key failure is something tolerable
 			glog.Warningf("Error parsing key %d (%d keys total) for %q: %v", i, numKeys, avt.Attestor.Name, err)
 			continue
 		}
-		if _, ok := keys[pubKey.KeyId]; ok {
-			glog.Warningf("Overwriting key with same fingerprint %s for %q.", pubKey.KeyId, avt.Attestor.Name)
+		// Base64 decode the attestor's key
+		decodedKey, err := base64.StdEncoding.DecodeString(attestorKey.AsciiArmoredPgpPublicKey)
+		if err != nil {
+			glog.Infof("Cannot base64 decode public key: %v", err)
+			continue
 		}
-		keys[pubKey.KeyId] = pubKey
+		publicKey := cryptolib.NewPublicKey(cryptolib.Pgp, decodedKey, attestorKey.KeyId)
+		publicKeys = append(publicKeys, publicKey)
 	}
-	if len(keys) == 0 {
+	if len(publicKeys) == 0 {
 		return nil, fmt.Errorf("unable to find any valid key for %q", avt.Attestor.Name)
 	}
 
-	out := []attestation.ValidatedAttestation{}
-	host, err := container.NewAtomicContainerSig(image, map[string]string{})
+	// Check for properly formatted image digest
+	digest, err := name.NewDigest(image, name.StrictValidation)
 	if err != nil {
-		glog.Error(err)
-		return nil, err
+		return nil, errors.Wrap(err, "error extracting image digest")
 	}
-	rawAtts, err := avt.Client.Attestations(image, &avt.Attestor)
+
+	verifier, err := cryptolib.NewVerifier(digest.DigestStr(), publicKeys)
 	if err != nil {
-		glog.Error(err)
-		return nil, err
+		return nil, errors.Wrap(err, "error creating verifier")
 	}
 
+	rawAtts, err := avt.Client.Attestations(image, &avt.Attestor)
```

Comment:
Do we plan to change this to return cryptolib.Attestation instead of RawAttestation?  Would that be a good idea?

(No need to do it now, I'm just curious.)

Reply:
We (me & @acamadeo ) discussed on this previously.
I think eventually `RawAttestation` could be completely replaced by `cryptolib.Attestation`. But due to `RawAttestation`'s large exposure surface, we'll for now do conversion between them and investigate replacement in later PRs.

Reply:
Yes, we will replace `RawAttestation` with `cryptolib.Attestation` in a future PR